sdk/internal/recording/server.go

```diff
@@ -0,0 +1,329 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package recording
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"runtime"
+	"time"
+)
+
+func getTestProxyDownloadFile() (string, error) {
+	if runtime.GOOS == "windows" {
+		return "test-proxy-standalone-win-x64.zip", nil
+	}
+
+	switch {
+	case runtime.GOOS == "linux" && runtime.GOARCH == "amd64":
+		return "test-proxy-standalone-linux-x64.tar.gz", nil
+	case runtime.GOOS == "linux" && runtime.GOARCH == "arm64":
+		return "test-proxy-standalone-linux-arm64.tar.gz", nil
+	case runtime.GOOS == "darwin" && runtime.GOARCH == "amd64":
+		return "test-proxy-standalone-osx-x64.zip", nil
+	case runtime.GOOS == "darwin" && runtime.GOARCH == "arm64":
+		return "test-proxy-standalone-osx-arm64.zip", nil
+	default:
+		return "", fmt.Errorf("unsupported OS/Arch combination: %s/%s", runtime.GOOS, runtime.GOARCH)
+	}
+}
+
+func extractTestProxyZip(archivePath string, outputDir string) error {
```

Comment:
Imagine if the test-proxy were just a single binary...

Reply:
It is a single binary, but we publish the release as an archive since it contains other things like debug symbols. @scbedd I wonder if we should also make the binaries themselves available on the releases page alongside the archives?

Reply:
I think that's a bit of work for not very much benefit.